lib/src/assignment_logger.dart

```diff
@@ -0,0 +1,68 @@
+/// Represents an event for logging an assignment
+class AssignmentEvent {
+  /// The allocation the subject was assigned to
+  final String? allocation;
+
+  /// The experiment identifier
+  final String? experiment;
+
+  /// The feature flag identifier
+  final String featureFlag;
+
+  /// The format of the assignment
+  final String format;
+
+  /// The variation key that was assigned
+  final String? variation;
+
+  /// The subject identifier
+  final String subject;
+
+  /// The timestamp of the assignment
+  final String timestamp;
```

Comment:
we've been inconsistent on whether we use native dates or strings in our SDK events. I'm slightly biased towards native dates.